platform/source/mbed_retarget.cpp

```diff
@@ -768,13 +861,31 @@ extern "C" int PREFIX(_read)(FILEHANDLE fh, unsigned char *buffer, unsigned int
 
 extern "C" ssize_t read(int fildes, void *buf, size_t length)
 {
+    if (length == 0) {
```

Comment:
I'd put this inside the ifdef, below the `fildes` check. You don't need it for the non-minimal case (because `FileHandle::read` is expected to get it right), and it should be after the checking the file handle is valid.